invenio_oauthclient/contrib/cern.py

```diff
@@ -185,9 +201,47 @@ def fetch_groups(groups):
     return groups
 
 
+def account_groups(account, resource, refresh_timedelta=None):
+    """Fetch account groups from resource if necessary."""
+    updated = datetime.utcnow()
+    modified_since = updated
+    if refresh_timedelta is not None:
+        modified_since += refresh_timedelta
+    modified_since = modified_since.isoformat()
+    last_update = account.extra_data.get('updated', modified_since)
+
+    if last_update > modified_since:
+        return account.extra_data.get('groups', [])
+
+    groups = fetch_groups(resource['Group'])
+    account.extra_data.update(
+        groups=groups,
+        updated=updated.isoformat(),
+    )
+    return groups
+
+
+def extend_identity(identity, groups):
+    """Extend identity with roles based on CERN groups."""
+    provides = set([UserNeed(current_user.email)] + [
+        RoleNeed('{0}@cern.ch'.format(name)) for name in groups
+    ])
+    identity.provides |= provides
+    session[OAUTHCLIENT_CERN_SESSION_KEY] = provides
+
+
+def disconnect_identity(identity):
```

Comment:
@jirikuncar it works good. I think we are done for now with this. Should we merge it  and create a new release?